src/Data/Fin/Properties.agda

```diff
@@ -679,13 +689,20 @@ combine-injective i j k l cᵢⱼ≡cₖₗ =
   combine-injectiveʳ i j k l cᵢⱼ≡cₖₗ
 
 combine-surjective : ∀ (i : Fin (m ℕ.* n)) → ∃₂ λ j k → combine j k ≡ i
-combine-surjective {m} {n} i with remQuot {m} n i | P.inspect (remQuot {m} n) i
-... | j , k | P.[ eq ] = j , k , (begin
+combine-surjective {m} {n} i with remQuot {m} n i in eq
+... | j , k = j , k , (begin
   combine j k                       ≡˘⟨ uncurry (cong₂ combine) (,-injective eq) ⟩
   uncurry combine (remQuot {m} n i) ≡⟨ combine-remQuot {m} n i ⟩
   i                                 ∎)
   where open ≡-Reasoning
 
+combine-surjectiveNEW : ∀ (i : Fin (m ℕ.* n)) → ∃₂ λ j k → combine {m} {n} j k ≡ i
```

Comment:
See my [earlier comment above](https://github.com/agda/agda-stdlib/pull/1930#issuecomment-1438534264): I provided two ways to prove the `combine-surjective` property, and had been waiting on a review judgment call as to which proof might be preferable

Reply:
I definitely like the old version more. Using other lemmas rather than building it up from first principles via pattern matching seems more robust to me.

Reply:
Ok, will clean up later today.